crescent/internal/registry.py

```diff
@@ -244,6 +244,9 @@ async def post_guild_command(self, commands: List[CommandBuilder], guild: Snowfl
             )
 
     async def register_commands(self):
+        if not self.bot.update_commands:
+            return
+
```

Comment:
It seems like it would be best to make `update_commands` determine whether or not this function is called, rather than making it disable this function alltogether. Either that, or make register_commands private.

Reply:
yeah that would be a lot better lol

Reply:
Turns out this is a massive PITA to write proper tests for

Reply:
Haha that's fine. Maybe do this instead:

```py
async def _register_if_enabled(self) -> None:
    if self.update_commands:
        await self.register_commands()
```

That way it's easy to write tests and the function is broken.